drivers/vulkan/vulkan_context.cpp

```diff
@@ -420,53 +425,148 @@ Error VulkanContext::_initialize_extensions() {
 #else
 	bool want_debug_utils = OS::get_singleton()->is_stdout_verbose();
 #endif
+	if (want_debug_utils) {
+		requested_extensions[VK_EXT_DEBUG_UTILS_EXTENSION_NAME] = &enabled_debug_utils;
+	} else {
+		enabled_debug_utils = false;
+	}
 
+	// TODO obtain requested extensions from wrappers/gdextensions if we decide to implement that model (see OpenXR module).
+
+	// Load instance extensions that are available...
+	uint32_t instance_extension_count = 0;
 	VkResult err = vkEnumerateInstanceExtensionProperties(nullptr, &instance_extension_count, nullptr);
 	ERR_FAIL_COND_V(err != VK_SUCCESS && err != VK_INCOMPLETE, ERR_CANT_CREATE);
+	ERR_FAIL_COND_V_MSG(instance_extension_count == 0, ERR_CANT_CREATE, "No instance extensions found, is a driver installed?");
 
-	if (instance_extension_count > 0) {
-		VkExtensionProperties *instance_extensions = (VkExtensionProperties *)malloc(sizeof(VkExtensionProperties) * instance_extension_count);
-		err = vkEnumerateInstanceExtensionProperties(nullptr, &instance_extension_count, instance_extensions);
-		if (err != VK_SUCCESS && err != VK_INCOMPLETE) {
-			free(instance_extensions);
-			ERR_FAIL_V(ERR_CANT_CREATE);
-		}
-		for (uint32_t i = 0; i < instance_extension_count; i++) {
-			if (!strcmp(VK_KHR_SURFACE_EXTENSION_NAME, instance_extensions[i].extensionName)) {
-				surfaceExtFound = 1;
-				extension_names[enabled_extension_count++] = VK_KHR_SURFACE_EXTENSION_NAME;
-			}
+	VkExtensionProperties *instance_extensions = (VkExtensionProperties *)malloc(sizeof(VkExtensionProperties) * instance_extension_count);
+	err = vkEnumerateInstanceExtensionProperties(nullptr, &instance_extension_count, instance_extensions);
+	if (err != VK_SUCCESS && err != VK_INCOMPLETE) {
+		free(instance_extensions);
+		ERR_FAIL_V(ERR_CANT_CREATE);
+	}
+#ifdef DEV_ENABLED
+	for (uint32_t i = 0; i < instance_extension_count; i++) {
+		print_verbose(String("VULKAN: Found instance extension ") + String(instance_extensions[i].extensionName));
+	}
+#endif
 
-			if (!strcmp(_get_platform_surface_extension(), instance_extensions[i].extensionName)) {
-				platformSurfaceExtFound = 1;
-				extension_names[enabled_extension_count++] = _get_platform_surface_extension();
-			}
-			if (!strcmp(VK_EXT_DEBUG_REPORT_EXTENSION_NAME, instance_extensions[i].extensionName)) {
-				if (_use_validation_layers()) {
-					extension_names[enabled_extension_count++] = VK_EXT_DEBUG_REPORT_EXTENSION_NAME;
-					enabled_debug_report = true;
-				}
-			}
-			if (!strcmp(VK_EXT_DEBUG_UTILS_EXTENSION_NAME, instance_extensions[i].extensionName)) {
-				if (want_debug_utils) {
-					extension_names[enabled_extension_count++] = VK_EXT_DEBUG_UTILS_EXTENSION_NAME;
-					enabled_debug_utils = true;
+	for (auto &requested_extension : requested_extensions) {
+		bool found = false;
+
+		for (uint32_t i = 0; !found && i < instance_extension_count; i++) {
```

Comment:
What kind of range can we expect from ``instance_extension_count`` if it is larger than a few entries we may want to avoid searching the entire array for each extension. Instead we could loop though ``instance_extensions`` and check if the ``requested_extension`` has ``extensionName``. Then loop through ``requested_extension`` once and verify that we got everything we needed

Reply:
Two dozen or so, it was the other way around originally but then you need to do an extra check for missing required extensions. I figured it's a once off thing we do that hardly registers as a blip in the startup process. I'd be concerned about it if we run this time and time again.

Reply:
Ok, changed how this works for instance extensions, will make the same change for device extensions. See top description for how it now works.